From 37cb88a1262bd9423cb767c9a9ac3250ba11ac0a Mon Sep 17 00:00:00 2001
From: Amit Shah <amit.shah@ti.com>
Date: Wed, 16 Nov 2011 13:47:20 +0530
Subject: [PATCH 0622/1092] ARM:OMAP:AM33XX: Add AM33XX_IRQ_GPMC0 as gpmc_irq
 for AM33XX

uio_pruss driver was unable to load due to gpmc driver bug wherein the gpmc_irq for AM33XX was being set to INT_34XX_GPMC_IRQ.
Signed-off-by: Amit Shah <amit.shah@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 arch/arm/mach-omap2/gpmc.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-omap2/gpmc.c b/arch/arm/mach-omap2/gpmc.c
index 4ee2e90..1cc4544 100644
--- a/arch/arm/mach-omap2/gpmc.c
+++ b/arch/arm/mach-omap2/gpmc.c
@@ -711,11 +711,13 @@ static int __init gpmc_init(void)
 		gpmc_irq = INT_34XX_GPMC_IRQ;
 	} else if (cpu_is_omap34xx()) {
 		ck = "gpmc_fck";
-		if (cpu_is_am33xx())
+		if (cpu_is_am33xx()) {
 			l = OMAP44XX_GPMC_BASE;
-		else
+			gpmc_irq = AM33XX_IRQ_GPMC0;
+		} else {
 			l = OMAP34XX_GPMC_BASE;
-		gpmc_irq = INT_34XX_GPMC_IRQ;
+			gpmc_irq = INT_34XX_GPMC_IRQ;
+		}
 	} else if (cpu_is_omap44xx()) {
 		ck = "gpmc_ck";
 		l = OMAP44XX_GPMC_BASE;
-- 
1.7.11.2

